pkg/dockerfile/generator.go

```diff
@@ -1,689 +1,19 @@
-// It returns the lines to add to Dockerfile to make it available and the filename it ends up as inside the container
-func (g *Generator) writeTemp(filename string, contents []byte) ([]string, string, error) {
-	path := filepath.Join(g.tmpDir, filename)
-	if err := os.MkdirAll(filepath.Dir(path), 0o755); err != nil {
-		return []string{}, "", fmt.Errorf("Failed to write %s: %w", filename, err)
-	}
-	if err := os.WriteFile(path, contents, 0o644); err != nil {
-		return []string{}, "", fmt.Errorf("Failed to write %s: %w", filename, err)
-	}
-	return []string{fmt.Sprintf("COPY %s /tmp/%s", filepath.Join(g.relativeTmpDir, filename), filename)}, "/tmp/" + filename, nil
-}
-
-func joinStringsWithoutLineSpace(chunks []string) string {
-	lines := []string{}
-	for _, chunk := range chunks {
-		chunkLines := strings.Split(chunk, "\n")
-		lines = append(lines, chunkLines...)
-	}
-	return strings.Join(filterEmpty(lines), "\n")
-}
-
-func filterEmpty(list []string) []string {
-	filtered := []string{}
-	for _, s := range list {
-		if s != "" {
-			filtered = append(filtered, s)
-		}
-	}
-	return filtered
-}
-
-func (g *Generator) GenerateWeightsManifest() (*weights.Manifest, error) {
-	m := weights.NewManifest()
-
-	for _, dir := range g.modelDirs {
-		err := g.fileWalker(dir, func(path string, info os.FileInfo, err error) error {
-			if err != nil {
-				return err
-			}
-			if info.IsDir() {
-				return nil
-			}
-
-			return m.AddFile(path)
-		})
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	for _, path := range g.modelFiles {
-		err := m.AddFile(path)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	return m, nil
-}
-
-func (g *Generator) determineBaseImageName() (string, error) {
-	var changed bool
-	var err error
-
-	cudaVersion := g.Config.Build.CUDA
-
-	pythonVersion := g.Config.Build.PythonVersion
-	pythonVersion, changed, err = stripPatchVersion(pythonVersion)
-	if err != nil {
-		return "", err
-	}
-	if changed {
-		console.Warnf("Stripping patch version from Python version %s to %s", g.Config.Build.PythonVersion, pythonVersion)
-	}
-
-	torchVersion, _ := g.Config.TorchVersion()
-
-	// validate that the base image configuration exists
-	imageGenerator, err := NewBaseImageGenerator(cudaVersion, pythonVersion, torchVersion)
-	if err != nil {
-		return "", err
-	}
-	baseImage := BaseImageName(imageGenerator.cudaVersion, imageGenerator.pythonVersion, imageGenerator.torchVersion)
-	return baseImage, nil
-}
-
-func stripPatchVersion(versionString string) (string, bool, error) {
-	if versionString == "" {
-		return "", false, nil
-	}
-
-	v, err := version.NewVersion(versionString)
-	if err != nil {
-		return "", false, fmt.Errorf("Invalid version: %s", versionString)
-	}
-
-	strippedVersion := fmt.Sprintf("%d.%d", v.Major, v.Minor)
-	changed := strippedVersion != versionString
-
-	return strippedVersion, changed, nil
+import "github.com/replicate/cog/pkg/weights"
+
+type Generator interface {
+	generateInitialSteps() (string, error)
```

Comment:
I think you can safely leave this method out of the interface 👍🏼 
```suggestion
```

Reply:
This is used by the base image generator here: https://github.com/replicate/cog/blob/main/pkg/dockerfile/base.go#L188

Reply:
Ah I totally missed that! In that case I think it should be made a public method `GenerateInitialSteps`.

Reply:
Resolved in [83769a2](https://github.com/replicate/cog/pull/2088/commits/83769a254c101e5017c2b7a9c6e75bf5c182d99a)